### PR TITLE
Cleanup actions to revert Jetpack-specific changes

### PR DESCRIPTION
**Pull request by AliSoftware**
This is part of project ref paaHJt-2sc-p2

 - [X] This PR removes all the instances of the `:app` parameter in the actions and helpers.
 - [X] Incidentally, it also fixes some warnings about `git commands` when running rspec locally (depending on the version of `git` you have installed on your system)

## Merge, Test and New Major Version

 - Before doing a new toolkit major version with those changes, I plan to test those changes with https://github.com/wordpress-mobile/WordPress-Android/pull/15441 during upcoming `finalize_release` and `code_freeze` of WPAndroid.
 - This PR can still be merged into `develop` once approved, even if we don't do a new major version until we've validated it in WPAndroid; we'll just need to be sure that `WPAndroid`'s `Pluginfile` points to `develop` (or better, to the `ref:` corresponding to the merge commit) once this PR is merged and the branch deleted.
 - Once we merge this PR and then WPAndroid confirm that those changes work, we should be ready to do a new major version for release-toolkit.

_In practice we could do a major version before the end of this sprint, and consider doing a minor or patch version afterwards if we find issues. But if the issues we find are breaking changes that means we'd need to bump to yet another major version, so I prefer to play it safe_

### CODE REVIEW

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_build_prechecks.rb [hunk 1]
```diff
@@ -9,16 +9,15 @@ def self.run(params)
 
         Fastlane::Helper::GitHelper.ensure_on_branch!('release') unless other_action.is_ci()
 
-        app = params[:app]
         message = ''
-        beta_version = Fastlane::Helper::Android::VersionHelper.get_release_version(product_name: app) unless !params[:beta] && !params[:final]
-        alpha_version = Fastlane::Helper::Android::VersionHelper.get_alpha_version(app) if params[:alpha]
+        beta_version = Fastlane::Helper::Android::VersionHelper.get_release_version() unless !params[:beta] && !params[:final]
+        alpha_version = Fastlane::Helper::Android::VersionHelper.get_alpha_version() if params[:alpha]
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_build_prechecks.rb]: Nitpick: Manual Rubocop 💅

```suggestion
        beta_version = Fastlane::Helper::Android::VersionHelper.get_release_version unless !params[:beta] && !params[:final]
        alpha_version = Fastlane::Helper::Android::VersionHelper.get_alpha_version if params[:alpha]
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_beta.rb [hunk 2]
```diff
@@ -8,22 +8,21 @@ def self.run(params)
         require_relative '../../helper/android/android_version_helper'
 
         Fastlane::Helper::GitHelper.ensure_on_branch!('release')
-        app = params[:app]
 
-        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version(product_name: app)
-        current_version_alpha = Fastlane::Helper::Android::VersionHelper.get_alpha_version(app)
+        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version()
+        current_version_alpha = Fastlane::Helper::Android::VersionHelper.get_alpha_version()
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_beta.rb]:
```suggestion
        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version
        current_version_alpha = Fastlane::Helper::Android::VersionHelper.get_alpha_version
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_beta.rb [hunk 3]
```diff
@@ -43,12 +42,6 @@ def self.details
 
       def self.available_options
         # Define all options your action supports.
-        [
-          FastlaneCore::ConfigItem.new(key: :app,
-                                       env_name: 'PROJECT_NAME',
-                                       description: 'The name of the app to get the release version for',
-                                       is_string: true), # true: verifies the input is a string, false: every kind of value
-        ]
       end
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_beta.rb:45]: Would it make sense to remove the entire method now that it effectively returns `nil`, as the one in the superclass [does](https://github.com/fastlane/fastlane/blob/27a41b83e1fd54ab70a040b3e6663dfe76449446/fastlane/lib/fastlane/action.rb#L53-L60).

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_final_release.rb [hunk 4]
```diff
@@ -8,20 +8,19 @@ def self.run(params)
         require_relative '../../helper/android/android_version_helper'
 
         Fastlane::Helper::GitHelper.ensure_on_branch!('release')
-        app = params[:app]
 
-        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version(product_name: app)
-        current_version_alpha = Fastlane::Helper::Android::VersionHelper.get_alpha_version(app)
+        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version()
+        current_version_alpha = Fastlane::Helper::Android::VersionHelper.get_alpha_version()
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_final_release.rb]:
```suggestion
        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version
        current_version_alpha = Fastlane::Helper::Android::VersionHelper.get_alpha_version
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_final_release.rb [hunk 5]
```diff
@@ -41,12 +40,6 @@ def self.details
 
       def self.available_options
         # Define all options your action supports.
-        [
-          FastlaneCore::ConfigItem.new(key: :app,
-                                       env_name: 'PROJECT_NAME',
-                                       description: 'The name of the app to get the release version for',
-                                       is_string: true), # true: verifies the input is a string, false: every kind of value
-        ]
       end
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_final_release.rb:43]: Duplicating from comment above: would it make sense to remove this, letting the superclass handle it?

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_hotfix.rb [hunk 6]
```diff
@@ -7,20 +7,18 @@ def self.run(params)
         require_relative '../../helper/android/android_git_helper'
         Fastlane::Helper::GitHelper.create_branch("release/#{params[:version_name]}", from: params[:previous_version_name])
 
-        app = params[:app]
-
-        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version(product_name: app)
+        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version()
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_hotfix.rb]:
```suggestion
        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_release.rb [hunk 7]
```diff
@@ -11,22 +11,21 @@ def self.run(params)
         other_action.ensure_git_branch(branch: 'develop')
 
         # Create new configuration
-        app = params[:app]
-        new_short_version = Fastlane::Helper::Android::VersionHelper.bump_version_release(app)
+        new_short_version = Fastlane::Helper::Android::VersionHelper.bump_version_release()
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_release.rb]:
```suggestion
        new_short_version = Fastlane::Helper::Android::VersionHelper.bump_version_release
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_release.rb [hunk 8]
```diff
@@ -11,22 +11,21 @@ def self.run(params)
...
 
-        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version(product_name: app)
-        current_version_alpha = Fastlane::Helper::Android::VersionHelper.get_alpha_version(app)
+        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version()
+        current_version_alpha = Fastlane::Helper::Android::VersionHelper.get_alpha_version()
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_release.rb]:
```suggestion
        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version
        current_version_alpha = Fastlane::Helper::Android::VersionHelper.get_alpha_version
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_release.rb [hunk 9]
```diff
@@ -54,12 +53,6 @@ def self.details
 
       def self.available_options
         # Define all options your action supports.
-        [
-          FastlaneCore::ConfigItem.new(key: :app,
-                                       env_name: 'PROJECT_NAME',
-                                       description: 'The name of the app to get the release version for',
-                                       is_string: true), # true: verifies the input is a string, false: every kind of value
-        ]
       end
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_release.rb:56]: Duplicating from comment above: would it make sense to remove this, letting the superclass handle it?

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_codefreeze_prechecks.rb [hunk 10]
```diff
@@ -14,18 +14,16 @@ def self.run(params)
         # Checkout develop and update
         Fastlane::Helper::GitHelper.checkout_and_pull('develop')
 
-        app = params[:app]
-
         # Create versions
-        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version(product_name: app)
-        current_alpha_version = Fastlane::Helper::Android::VersionHelper.get_alpha_version(app)
+        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version()
+        current_alpha_version = Fastlane::Helper::Android::VersionHelper.get_alpha_version()
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_codefreeze_prechecks.rb]:
```suggestion
        current_version = Fastlane::Helper::Android::VersionHelper.get_release_version
        current_alpha_version = Fastlane::Helper::Android::VersionHelper.get_alpha_version
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_codefreeze_prechecks.rb [hunk 11]
```diff
@@ -37,7 +35,7 @@ def self.run(params)
         other_action.ensure_git_status_clean()
 
         # Return the current version
-        Fastlane::Helper::Android::VersionHelper.get_public_version(app)
+        Fastlane::Helper::Android::VersionHelper.get_public_version()
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_codefreeze_prechecks.rb]:
```suggestion
        Fastlane::Helper::Android::VersionHelper.get_public_version
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_completecodefreeze_prechecks.rb [hunk 12]
```diff
@@ -9,10 +9,8 @@ def self.run(params)
 
         UI.user_error!('This is not a release branch. Abort.') unless other_action.git_branch.start_with?('release/')
 
-        app = params[:app]
-
-        version = Fastlane::Helper::Android::VersionHelper.get_public_version(app)
-        message = "Completing code freeze for: [#{app}]#{version}\n"
+        version = Fastlane::Helper::Android::VersionHelper.get_public_version()
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_completecodefreeze_prechecks.rb]:
```suggestion
        version = Fastlane::Helper::Android::VersionHelper.get_public_version
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_current_branch_is_hotfix.rb [hunk 13]
```diff
@@ -7,7 +7,7 @@ module SharedValues
     class AndroidCurrentBranchIsHotfixAction < Action
       def self.run(params)
         require_relative '../../helper/android/android_version_helper'
-        version = Fastlane::Helper::Android::VersionHelper.get_release_version(product_name: params[:app])
+        version = Fastlane::Helper::Android::VersionHelper.get_release_version()
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_current_branch_is_hotfix.rb]:
```suggestion
        version = Fastlane::Helper::Android::VersionHelper.get_release_version
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_current_branch_is_hotfix.rb [hunk 14]
```diff
@@ -25,12 +25,6 @@ def self.details
 
       def self.available_options
         # Define all options your action supports.
-        [
-          FastlaneCore::ConfigItem.new(key: :app,
-                                       env_name: 'PROJECT_NAME',
-                                       description: 'The name of the app to get the release version for',
-                                       is_string: true), # true: verifies the input is a string, false: every kind of value
-        ]
       end
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_current_branch_is_hotfix.rb:28]: Duplicating from comment above: would it make sense to remove this, letting the superclass handle it?

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_finalize_prechecks.rb [hunk 15]
```diff
@@ -13,8 +13,8 @@ def self.run(params)
 
         UI.user_error!('This is not a release branch. Abort.') unless other_action.git_branch.start_with?('release/')
 
-        version = Fastlane::Helper::Android::VersionHelper.get_public_version(params[:app])
-        message = "Finalizing #{params[:app]} release: #{version}\n"
+        version = Fastlane::Helper::Android::VersionHelper.get_public_version()
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_finalize_prechecks.rb]:
```suggestion
        version = Fastlane::Helper::Android::VersionHelper.get_public_version
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_get_alpha_version.rb [hunk 16]
```diff
@@ -3,7 +3,7 @@ module Actions
     class AndroidGetAlphaVersionAction < Action
       def self.run(params)
         require_relative '../../helper/android/android_version_helper'
-        Fastlane::Helper::Android::VersionHelper.get_alpha_version(params[:app])
+        Fastlane::Helper::Android::VersionHelper.get_alpha_version()
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_get_alpha_version.rb]:
```suggestion
        Fastlane::Helper::Android::VersionHelper.get_alpha_version
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_get_alpha_version.rb [hunk 17]
```diff
@@ -20,12 +20,6 @@ def self.details
 
       def self.available_options
         # Define all options your action supports.
-        [
-          FastlaneCore::ConfigItem.new(key: :app,
-                                       env_name: 'PROJECT_NAME',
-                                       description: 'The name of the app to get the release version for',
-                                       is_string: true), # true: verifies the input is a string, false: every kind of value
-        ]
       end
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_get_alpha_version.rb:23]: Duplicating from comment above: would it make sense to remove this, letting the superclass handle it?

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_get_app_version.rb [hunk 18]
```diff
@@ -3,7 +3,7 @@ module Actions
     class AndroidGetAppVersionAction < Action
       def self.run(params)
         require_relative '../../helper/android/android_version_helper'
-        Fastlane::Helper::Android::VersionHelper.get_public_version(params[:app])
+        Fastlane::Helper::Android::VersionHelper.get_public_version()
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_get_app_version.rb]:
```suggestion
        Fastlane::Helper::Android::VersionHelper.get_public_version
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_get_app_version.rb [hunk 19]
```diff
@@ -20,12 +20,6 @@ def self.details
 
       def self.available_options
         # Define all options your action supports.
-        [
-          FastlaneCore::ConfigItem.new(key: :app,
-                                       env_name: 'PROJECT_NAME',
-                                       description: 'The name of the app to get the release version for',
-                                       is_string: true), # true: verifies the input is a string, false: every kind of value
-        ]
       end
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_get_app_version.rb:23]: Duplicating from comment above: would it make sense to remove this, letting the superclass handle it?

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_get_release_version.rb [hunk 20]
```diff
@@ -3,7 +3,7 @@ module Actions
     class AndroidGetReleaseVersionAction < Action
       def self.run(params)
         require_relative '../../helper/android/android_version_helper'
-        Fastlane::Helper::Android::VersionHelper.get_release_version(product_name: params[:app])
+        Fastlane::Helper::Android::VersionHelper.get_release_version()
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_get_release_version.rb]:
```suggestion
        Fastlane::Helper::Android::VersionHelper.get_release_version
```

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_get_release_version.rb [hunk 21]
```diff
@@ -20,12 +20,6 @@ def self.details
 
       def self.available_options
         # Define all options your action supports.
-        [
-          FastlaneCore::ConfigItem.new(key: :app,
-                                       env_name: 'PROJECT_NAME',
-                                       description: 'The name of the app to get the release version for',
-                                       is_string: true), # true: verifies the input is a string, false: every kind of value
-        ]
       end
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_get_release_version.rb:23]: Duplicating from comment above: would it make sense to remove this, letting the superclass handle it?

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_tag_build.rb [hunk 22]
```diff
@@ -5,10 +5,8 @@ def self.run(params)
         require_relative '../../helper/android/android_version_helper'
         require_relative '../../helper/android/android_git_helper'
 
-        app = ENV['PROJECT_NAME'].nil? ? params[:app] : ENV['PROJECT_NAME']
-
-        release_ver = Fastlane::Helper::Android::VersionHelper.get_release_version(product_name: app)
-        alpha_ver = Fastlane::Helper::Android::VersionHelper.get_alpha_version(app)
+        release_ver = Fastlane::Helper::Android::VersionHelper.get_release_version()
+        alpha_ver = Fastlane::Helper::Android::VersionHelper.get_alpha_version()
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_tag_build.rb]:
```suggestion
        release_ver = Fastlane::Helper::Android::VersionHelper.get_release_version
        alpha_ver = Fastlane::Helper::Android::VersionHelper.get_alpha_version
```

File: lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_version_helper.rb [hunk 23]
```diff
@@ -26,15 +26,12 @@ module VersionHelper
...
-        #                     No longer used, will be removed in future breaking update
-        #
         # @return [String] The public-facing version number, extracted from the `versionName` of the `build.gradle` file.
         #         - If this version is a hotfix (more than 2 parts and 3rd part is non-zero), returns the "X.Y.Z" formatted string
         #         - Otherwise (not a hotfix / 3rd part of version is 0), returns "X.Y" formatted version number
         #
-        def self.get_public_version(app)
-          version = get_release_version(product_name: app)
+        def self.get_public_version
+          version = self.get_release_version
```
Comment (mokagio) [inline lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_version_helper.rb]: I'm curious about this change. From what I make of the diff, calling with `self.` was not required before and `get_release_version` did not change from instance to class method either.

I'm guessing you made this change for consistency? To make it clearer to the reader `get_release_version` is a class method, in case they didn't notice it's being called from within one?

**Review by mokagio — COMMENTED** [review]
I run out of time to test this in the WordPress Android counterpart.

Submitting my original comments in the mean time.

File: lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_version_helper.rb (hunk 23 above)
Comment (AliSoftware) [reply lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_version_helper.rb]: tbh I actually have no recollection of why I did that change 😅 most likely it came about during some mass search/replace pass or something… but no real rationale behind it 🙃

I can revert it to not pollute the diff.

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_beta.rb (hunk 3 above)
Comment (AliSoftware) [reply lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_bump_version_beta.rb:45]: I'm not sure. It's all a matter of taste but we've always kept that method around in the actions even if empty in all the other actions in the toolkit, for two reasons:

 - It makes it explicit when reading the action code that the action doesn't take any parameter
 - It makes it easier to add new ones when needed, always at the same position in the action files

Also, the method is generated for us when you create a new plugin via the appropriate fastlane command (in other words, it's part of the official template for creating fastlane actions)

At the end of the day it's probably mostly a matter of style, conventions and preferences, so I'm good with either but since we've kept those everywhere else in the toolkit codebase, I'd prefer to keep them there too for consistency.

File: lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_version_helper.rb (hunk 23 above)
Comment (AliSoftware) [reply lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_version_helper.rb]: Fixed in c25108d830c06517eba1edf0f772332063e1ee61

File: lib/fastlane/plugin/wpmreleasetoolkit/actions/common/create_release_action.rb [hunk 24]
```diff
@@ -10,7 +10,7 @@ def self.run(params)
         repository = params[:repository]
         version = params[:version]
         assets = params[:release_assets]
-        release_notes = params[:release_notes_file_path].nil? ? '' : IO.read(params[:release_notes_file_path])
+        release_notes = params[:release_notes_file_path].nil? ? '' : File.read(params[:release_notes_file_path])
```
Comment (AliSoftware) [inline lib/fastlane/plugin/wpmreleasetoolkit/actions/common/create_release_action.rb:13]: This change is not directly related to that PR's topic but was flagged as a [security issue](https://docs.rubocop.org/rubocop/cops_security.html#securityiomethods) when I ran `rubocop`, so decided to take the occasion to fix it while at it. See 4885e0fc55f445e82a30f9eeb875f0ada9e5118c.

**Review by mokagio — APPROVED** [review]
Approving after testing via the Android PR using these updated actions: https://github.com/wordpress-mobile/WordPress-Android/pull/15441#pullrequestreview-778017725.

**Comment by AliSoftware** [comment]
Note for future reference: Buildkite CI failures on this PR are expected: the pipeline is enabled on Buildkite dashboard but the PR creating the `.buildkite/pipeline.yml` (#299) isn't merged yet
